### PR TITLE
Fix modal video tagging

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Fixes modal video tagging

https://github.com/user-attachments/assets/801cbc1c-18e9-4b06-9709-97f67bbe1bcc

## How is this patch tested? If it is not, please explain why.

Unit test for sample node updates in the grid

## Release Notes

* Fixed tagging in the modal for video samples

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **Refactor**
	- Streamlined how sample data is processed for improved reliability and easier maintenance.
	- Simplified error handling to ensure smoother operations.
	- Enhanced organization of modal configuration settings for better clarity.
- **New Features**
	- Introduced a new utility function for handling node processing.
- **Tests**
	- Introduced automated tests to verify consistent and correct handling of node processing.

### CODE REVIEW

File: app/packages/core/src/components/Grid/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it("handles object sample nodes", () => {
+    expect(handleNode(NODE)).toStrictEqual(NODE);
+  });
+
+  it("handles string sample nodes", () => {
+    // @ts-ignore
+    const node = handleNode({ ...NODE, sample: "{}" });
+    expect(node).toStrictEqual(NODE);
+  });
+});
```
[the remark this answers is not in the document]
Comment (tom-vx51) [reply app/packages/core/src/components/Grid/utils.test.ts]: agreed would recommend adding a test to cover the negative case

**Review by tom-vx51 — APPROVED** [review]

**Review by tom-vx51 — APPROVED** [review]

**Review by jleven — APPROVED** [review]
LGTM -- but you should probably get a second opinion
